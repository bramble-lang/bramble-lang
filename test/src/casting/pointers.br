fn my_main() -> i64 {
    let i: i64 := (18441921395520346504u64) as i64;
    let p: *const i64 := @const i;
    let bp: *const u8 := p as *const u8;

    project::std::io::writeu8ln(^(bp@0));
    project::std::io::writeu8ln(^(bp@1));
    project::std::io::writeu8ln(^(bp@2));
    project::std::io::writeu8ln(^(bp@3));
    project::std::io::writeu8ln(^(bp@4));
    project::std::io::writeu8ln(^(bp@5));
    project::std::io::writeu8ln(^(bp@6));
    project::std::io::writeu8ln(^(bp@7));
    return 0;
}

struct MyStruct {
    i: i64,
    j: i64,
    k: i64,
}