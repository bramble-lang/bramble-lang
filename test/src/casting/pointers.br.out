136
153
170
187
204
221
238
255
